src/api/process/process.rs

```diff
@@ -39,11 +43,11 @@ pub enum MemoryChoice {
 
 /// A lunatic process represents an actor.
 pub struct Process {
-    task: Task<Result<()>>,
+    task: Task<Result<HeapProfilerState>>,
```

Comment:
We can leave it for now but we will revert to `()` as host functions state will persist after a process halts.

Reply:
I will do another small PR where I will remove it and where I will remove merging profilers as it is done in the current form (so that processes can be observed as suggested by you here https://github.com/lunatic-solutions/lunatic/pull/44#discussion_r626264324)